rom/dev/src/main.rs

```diff
@@ -95,9 +95,10 @@ pub extern "C" fn rom_entry() -> ! {
     cprintln!("[state] LifecycleState = {}", _lifecyle);
 
     if cfg!(feature = "fake-rom")
-        && env.soc_ifc.lifecycle() == caliptra_drivers::Lifecycle::Production
+        && (env.soc_ifc.lifecycle() == caliptra_drivers::Lifecycle::Production)
```

Comment:
What is the motivation for supporting the Production lifecycle mode for the fake ROM?

Reply:
The motivation is just that more exhaustive testing in simulation was needed than I think we originally anticipated. This includes several test lines in production mode. (I included some more details in my response below.)